include/boost/geometry/algorithms/detail/overlay/get_turn_info_la.hpp

```diff
@@ -797,12 +802,33 @@ struct get_turn_info_linear_areal
             }
             else //if ( result.template get<0>().count == 1 )
             {
-                int const side_pi_q2 = sides.apply(range_q.at(1), range_q.at(2), range_p.at(0));
-                int const side_pi_x = sides.apply(range_q.at(0), range_p.at(1), range_p.at(0));
-                int const side_qk_x = sides.apply(range_q.at(0), range_p.at(1), range_q.at(2));
+                // pj is the intersection point at qj or in the middle of q1
+                // so consider segments
+                // 1. pj at qj: qi-qj-pi and qi-qj-qk
+                //    x: qi-qj, y: qj-qk, qz: qk
+                // 2. pj in the middle of q1: qi-pj-pi and qi-pj-qj
+                //    x: qi-pj, y: pj-qj, qz: qj
+                //    qi-pj, the side is the same as WRT q1
+                //    pj-qj, the side is the same as WRT q1
+                //    side of qj WRT q1 is 0
+                int side_pi_y = 0, side_pi_x = 0, side_qz_x = 0;
+                // 1. ip0 or pj at qj
+                if ( ip0.is_qj )
+                {
+                    side_pi_y = sides.apply(range_q.at(1), range_q.at(2), range_p.at(0)); // pi wrt q2
+                    side_pi_x = sides.apply(range_q.at(0), range_q.at(1), range_p.at(0)); // pi wrt q1
+                    side_qz_x = sides.apply(range_q.at(0), range_q.at(1), range_q.at(2)); // qk wrt q1   
+                }
+                // 2. ip0 or pj in the middle of q1
+                else
+                {
+                    side_pi_y = sides.apply(range_q.at(0), range_q.at(1), range_p.at(0)); // pi wrt q1
+                    side_pi_x = side_pi_y; // pi wrt q1
```

Comment:
Great, so all sides are real sides now, a point w.r.t. an existing segment.
Much better and better documented and named.
I'm glad you worked on this last bit, you know and understand this code much better than I do.